### ISSUE TITLE
CA1307 false-positive on Regex.Replace

### ISSUE THREAD

**Issue by geekley** [dotnet/roslyn-analyzers#7151]
### Analyzer

**Diagnostic ID**: [CA1307](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1307): `Specify StringComparison for clarity`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 8 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.101](https://packages.ubuntu.com/mantic-updates/dotnet8)

### Describe the bug

I'm getting a false positive for CA1307 when using `string Regex.Replace(string input, MatchEvaluator evaluator)`.
It probably shouldn't trigger on regexes at all, even more so in my case where the regex already has the flag `RegexOptions.CultureInvariant`.
The warning message seems to indicate the analyzer is not properly checking the method signature:
> 'string.Replace(string, string)' has a method overload that takes a 'StringComparison' parameter. Replace this call in '[...]' with a call to 'string.Replace(string, string, System.StringComparison)' for clarity of intent.

## Additional context

It happens when setting `<TargetFramework>netstandard2.1</TargetFramework>` in the csproj, but not on `netstandard2.0`.
Some of my csproj settings:
```xml
<LangVersion>8.0</LangVersion>
<Nullable>warnings</Nullable>
<EnableNETAnalyzers>true</EnableNETAnalyzers>
<AnalysisLevel>latest-All</AnalysisLevel>
```
The code looks like this:
```cs
myRegex.Replace(str, match => ...)
```

**Comment by geekley** [comment]
Nevermind, this bug is invalid. My bad.
